### PR TITLE
Add batch transactions method

### PR DESCRIPTION
**Pull request by benmercerdev**
Adds a Batch Transactions class

[ch17403]

### CODE REVIEW

File: lib/synapse_pay_rest.rb [hunk 1]
```diff
@@ -77,6 +77,7 @@
 
 # transaction-related classes
 require 'synapse_pay_rest/models/transaction/transaction'
+require 'synapse_pay_rest/models/transaction/batch_transaction'
```
Comment (benmercerdev) [inline lib/synapse_pay_rest.rb:80]: good, thanks

File: lib/synapse_pay_rest/api/transactions.rb [hunk 2]
```diff
@@ -109,8 +109,8 @@ def delete(user_id:, node_id:, trans_id:)
     #
     # @return [Hash] API response
 
-    def create_batch_transaction(user_id:, node_id:, payload:, idempotency_key: nil)
-      path = create_batch_transaction_path(user_id: user_id, node_id: node_id, batch_id: batch_id)
+    def create_batch(user_id:, node_id:, payload:, idempotency_key: nil)
```
Comment (benmercerdev) [inline lib/synapse_pay_rest/api/transactions.rb:112]: good rename

File: lib/synapse_pay_rest/models/transaction/transaction.rb [hunk 3]
```diff
@@ -152,6 +151,44 @@ def from_response(node, response)
 
       private
 
+      def payload_for_create(node:, to_type:, to_id:, amount:, currency:, ip:, **options)
```
Comment (benmercerdev) [inline lib/synapse_pay_rest/models/transaction/transaction.rb:154]: Curious about the reasoning to pull this out of the `PayloadCreator` and add this back in here? Both the BatchTransaction and Transaction classes will need to use the same function

File: lib/synapse_pay_rest/models/transaction/batch_transaction.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+        }
+        self.new(**args)
+      end
+
+      private
+
+      def payload_for_batch_create(transactions:, **options)
+        {
+          'transactions' => transactions.map{|transaction| payload_for_create(transaction)}
```
Comment (benmercerdev) [inline lib/synapse_pay_rest/models/transaction/batch_transaction.rb]: @hendragunz I see you renamed this method back to `payload_for_create` and removed the `PayloadCreator` import -- the reason behind the `PayloadCreator` was to share the `payload_for_single_transaction` function in both the `Transaction` and `BatchTransaction` classes

File: lib/synapse_pay_rest/models/batch_transactions.bak/batch_trans.rb [hunk 5]
```diff
@@ -0,0 +1,85 @@
+# module SynapsePayRest
```
Comment (carltonsmith) [inline lib/synapse_pay_rest/models/batch_transactions.bak/batch_trans.rb]: Seems like this file should be deleted?

**Review by carltonsmith — CHANGES_REQUESTED** [review]
Aside from the file that looks like it should be removed, this looks good

File: lib/synapse_pay_rest/models/batch_transactions.bak/batch_trans.rb (hunk 5 above)
Comment (hendragunz) [reply lib/synapse_pay_rest/models/batch_transactions.bak/batch_trans.rb]: Oh.. yeah I missed that

**Review by carltonsmith — APPROVED** [review]

**Comment by hendragunz** [comment]
@carltonsmith @benmercerdev  just info.. actually this PR can be ignored. 
Because those all changes are embedded directly in GigWage application. 

I put it into `lib/synapse_pay_rest/monkey_pathces`. 
You can check there

**Comment by carltonsmith** [comment]
@hendragunz We are in the midst of moving all of the monkey patch changes into the gem, so I would rather we have it here instead

**Comment by hendragunz** [comment]
> @hendragunz We are in the midst of moving all of the monkey patch changes into the gem, so I would rather we have it here instead

will move those codes into SynapsePayRest gem instead @carltonsmith

**Pull request merged**